### DISCUSSION TITLE
WritableVariable access

### DISCUSSION THREAD

**Discussion by Flolaffel** [idaholab/moose#26816] [Q&A General] [answered]
Hello,

I'm using UserObjects to read from and write to AuxVariables. For writing purposes I use WritableVariables. The Problem I'm facing is following error

```
*** ERROR ***
'calc_sense' already obtained a writable reference to 'dV'. Only one object can obtain such a reference per variable and subdomain in a simulation.
```

The workflow of the simulation is:
1. Calculate the initial value of a scalar and an elemental AuxVariable in a UserObject and write them. The elemental AuxVariable uses a WritableVariable reference.
2. Read the value from the elemental AuxVariable in a second UserObject, do some modifications and write it back using a WritableVariable reference once again (which is where the error occurs).
3. Read both the scalar and the elemental AuxVariable again for the final computation.

Is there a way to get this working as it is? Or do I have to split my UserObject into an AuxScalarKernel and an elemental AuxKernel?

**Comment by GiudGiud** [comment, marked as answer]
Hello

This check was added on purpose to avoid execution ordering issues. 

The cleanest way forward here would be to turn one of your user objects into auxiliary kernels. These are the cannon way of setting auxiliary variables.

If you can't do that, I d just comment out the check in the framework and make sure things are still computed in the expected order 

Guillaume

**Reply by Flolaffel** [reply]
Ok thanks. I'll contemplate the switch to AuxKernels since that would also resolve the other issue I'm having  (#26806). I just thought it'd be neat to compute the scalar value and the matching elemental value in one object.

I'm controlling the execution order through setting the execution order group, so there shouldn't be any issues there.

**Reply by Flolaffel** [reply]
For now I decided to just go with the "comment out" option so I don't have to refactor my code.
